### ISSUE TITLE
How to compare loops of Hi-C from different conditions like DEG in RNA-Seq?

### ISSUE THREAD

**Issue by Linhua-Sun** [koszullab/chromosight#47]
Hi:
Thanks for your great chromosight! We have used chromosight widely in a plant's Hi-C. Now, I am wondering how to compare the loops？For example, if we got two Hi-C samples under different conditions.
Option 1: we can call loops for A and B Hi-C separately. Then we can merged the LoopA and LoopB (sometimes, loops results differ too much which is inconsistent with hic results in juicebox ). Then we can use chromosight quantify to calculate similarity to kernel. Just DEG analysis in RNA-Seq, can we identify those significantly gain/loss loops in different conditions?
Option 2: we can merge the two hic files and call loops in an artificial merged hic samples. 
In summary, how to call loops for two or more Hi-Cs and how to compare them?
Thank you for your time.
Best wishes!
Linhua

**Comment by cmdoret** [comment]
Hi @Linhua-Sun ,

Glad you're enjoying the software !

That's a good question, I usually go with option 2: Merge all samples -> `detect` to get a list of positions common to all samples, and then use it to `quantify` in each sample separately. This will yield a value for each sample on the same set of positions and you can compare between conditions using multiple replicates. Unfortunately, unless you have many replicates, it is hard to define "DEG loops", so you'll most likely have to apply a threshold.

When you do this, you need to be careful that all samples have ~the same coverage (`--subsample` flag is useful here), otherwise higher coverage samples will yield better loops.

I am working on a tool ([koszullab/pareidolia](https://github.com/koszullab/pareidolia)) to detect significantly different patterns between conditions, but this is still very much a work in progress and unstable.

**Comment by Linhua-Sun** [comment]
Hi:
Thank you very much for your quick reply. The information you provided is very important. I am looking forward to your new tools.
By the way, we got two replicates for each conditions, is it OK to perform the "DEG loops" analysis?
Thanks again!

**Comment by cmdoret** [comment]
This is a bit tricky, for RNAseq it was proposed that the minimum _should be_ 6 replicates, and 12 to get a good statistical power (source: https://pubmed.ncbi.nlm.nih.gov/27022035/). In practice, you almost never see that many replicates due to cost limitations.

I think we face the same issue with Hi-C; there are tools for differential Hi-C analysis based on edgeR, such as [diffHiC](https://bioconductor.org/packages/release/bioc/html/diffHic.html). There is also [CHESS](https://github.com/vaquerizaslab/chess) that works well. These tools may identify regions with "significantly" different contacts (they are agnostic to patterns and only look at contact increase / decrease).

One thing you could do is to run these tools and then see if you have Chromosight loops that fall in these differential regions. But with 2 replicates, only regions with very strong changes will be "statistically significant". So basically you will detect loops that completely disappear, but not subtle changes which could be important for regulation. Detection of these "subtle change" (dubbed 3D QTL) has been done previously with 11 replicates (source: https://link.springer.com/article/10.1186/s13059-019-1855-4).

I saw [this method](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-02061-9) which claims to detect differential loops even with 2 replicates, but tbh I have never managed to get it to work...

Alternatively, you could try to directly compare loop scores as I mentioned previously using Chromosight (or try using pareidolia, which essentially does the same thing), knowing you will have no statistically significant results due to sample size. If you already have genes or regions you're interested in, you could still check the score difference between conditions, and where it falls in the distribution. For now, what I do in pareidolia is applying a signal-to-noise ratio (replicate variance < condition variance) and a percentile threshold.

**Comment by Linhua-Sun** [comment]
Thank you very much for your detailed reply. Your reply has greatly improved our understanding. We will further test it based on your suggestions. At present, perhaps the best way is to directly calculate the value of each loop, and find a hard suitable threshold.
We are also testing the pareidolia.
Thanks again!
